PyGame_Project/MVC/View_GUI/screens/game_screen_components/pop_ups/popups.py

```diff
@@ -334,3 +336,174 @@ def on_yes(self, input_text=None):
 
     def on_no(self):
         self.__init__(self.state)
+
+
+class HighScorePopup(Popup):
+    def __init__(self, state):
+        self.state = state
+        self.required_letter = ''
+        self.puzzle_letters = ''
+        self.message = "Enter a puzzle to look up:"
+
+        self.confirmation_bool = False
+        self.valid_required_letter = False
+
+        super().__init__(state.display, self.message, self.on_yes, self.on_no)
+        self.no_button_text = "Cancel"
+        self.yes_button_text = "Confirm"
+
+    def on_yes(self):
+        if not self.confirmation_bool:
+            if len(self.text_input) == 7 and len(self.text_input) == len(set(self.text_input)):
```

Comment:
Hmm, I was under the impression when looking up a highscore it was whatever 7 letters, but I guess it would make sense that it works the same way as starting from a base word.